src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -329,6 +329,17 @@ private static ProjectRestoreMetadata GetMSBuildMetadata(PackageSpec packageSpec
                 msbuildMetadata.ProjectWideWarningProperties = new WarningProperties(warnAsError, noWarn, allWarningsAsErrors);
             }
 
+            // read NuGet lock file msbuild properties
+            var restoreLockProperties = rawMSBuildMetadata.GetValue<JObject>("restoreLockProperties");
```

Comment:
nit: I know you are following a pattern here, but having a constant would help avoiding weird bugs like case inconsistency.

Reply:
It's up to the author who is adding new property needs to make sure to use the correct casing, even with constant, we can fault into that issue. Also, it will be little weird with constant since then it will be something like: 

public static readonly string RestoreLockProperties = "restoreLockProperties";